### DISCUSSION TITLE
wire:model not working on modal

### DISCUSSION THREAD

**Discussion by yaj123** [livewire/livewire#8345] [1. Help]
Hello,

I'm new to laravel live:wire, so I'm trying to make a default value of the field on modal edit form. .but wire:model don't work. . 

Here is my livewire/UserTable

 use WithPagination;
    public $selectedUser = 0;
    public $selectedUsername,$selectedName,$selectedLocation,$selectedPosition;
    public $selectedNumber;
    public $number;
    public $userObject;
    
    
    
    public $listeners = [
        'refreshTable' => 'render',
    ];

    
    public function showEditModal($userID){
        $this->userObject = User::find($userID);
        $this->selectedNumber = $this -> userObject -> number;
        $this->selectedUsername = $this -> userObject -> username;
        $this->selectedName = $this -> userObject -> name;
        $this->selectedLocation = $this -> userObject -> location;
        $this->selectedPosition = $this -> userObject -> position;
        
      
    }
    
    public function mount()
    {
        $this->number = $this->selectedNumber;
    }
    
    
    
Here is my user-edit.blade.php

<div wire:ignore.self id="editUserModal" tabindex="-1" aria-hidden="true" class="modal fade fixed top-0 left-0 right-0 z-50 items-center justify-center hidden w-full p-4 overflow-x-hidden overflow-y-auto md:inset-0 h-[calc(100%-1rem)] max-h-full">
    
    <div class="relative w-full max-w-2xl max-h-full">
        
        <form class="relative bg-white rounded-lg shadow dark:bg-gray-700">
            @csrf
            
            <div class="flex items-start justify-between p-4 border-b rounded-t dark:border-gray-600">
                <h3 class="text-xl font-semibold text-gray-900 dark:text-white">
                    Edit user
                </h3>
            <button type="button" class="text-gray-400 bg-transparent hover:bg-gray-200 hover:text-gray-900 rounded-lg text-sm w-8 h-8 ms-auto inline-flex justify-center items-center dark:hover:bg-gray-600 dark:hover:text-white" data-modal-hide="editUserModal">
                <svg class="w-3 h-3" aria-hidden="true" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 14 14">
                    <path stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="m1 1 6 6m0 0 6 6M7 7l6-6M7 7l-6 6"/>
                </svg>
                <span class="sr-only">Close modal</span>
            </button>
            </div>
            
            <div class="p-6 space-y-6">
                <div class="grid grid-cols-6 gap-6">
                    <div class="col-span-6 sm:col-span-3">
                        <x-input-label for="username">Username</x-input-label>
                        <x-text-input type="text" name="username" id="username" placeholder="Username" required="" value="{{ $selectedUsername }}"/>
                    </div>
                    <div class="col-span-6 sm:col-span-3">
                        <x-input-label for="name">Name</x-input-label>
                        <x-text-input type="text" name="name" id="name" placeholder="Name" required="" value="{{ $selectedName}}" />
                    </div>
                    <div class="col-span-6 sm:col-span-3">
                        <x-input-label for="location">Location</x-input-label>
                        <x-text-input type="text" name="location" id="location" placeholder="Location" required="" value="{{ $selectedLocation }}"/>
                    </div>
                    <div class="col-span-6 sm:col-span-3">
                        <x-input-label for="number">Number</x-input-label>
                        <x-text-input wire:model.live="number" type="text" oninput="this.value = this.value.replace(/[^0-9.]/g, '').replace(/(\..*?)\..*/g, '$1');" name="number" id="number" placeholder="Number" required="" maxlength="11" />
                    </div>
                    <div class="col-span-6 sm:col-span-3">
                        <x-input-label for="position">Position</x-input-label>
                        <x-text-input type="text" name="position" id="position" placeholder="Position" required="" value="{{ $selectedPosition }}"/>
                    </div>
                    <div class="col-span-6 sm:col-span-3">
                        <x-input-label for="password">New Password</x-input-label>
                        <x-text-input type="password" name="password" id="password" placeholder="••••••••"/>
                    </div>   
                </div>
            </div>
            
            <div class="flex items-center p-6 space-x-3 rtl:space-x-reverse border-t border-gray-200 rounded-b dark:border-gray-600">
                <button wire:click="userUpdate" type="submit" class="text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800">Save</button>
            </div>
        </form>
    </div>
</div>

I can set the default value if I'm just using the value="{{$value}}" but with wire:model, it won't work..

Help..

i also tried directly assigning in showEditModal with $this->number = $this->userObject->number; but still don't work..

**Comment by tranKILL** [comment]
Check the validation doc : https://livewire.laravel.com/docs/validation

**Reply by yaj123** [reply]
Hello I'm still in filling up the field with default values. but wire:model won't work.
as you can see. i can assign default values with value={{$value}} but in the number field. I'm using the wire:model which won't work.

![Capture](https://github.com/livewire/livewire/assets/18673818/704bf404-bde0-483e-8e6c-83b84fcbf9d6)

and also I need the wire:model for the validation..

**Reply by tranKILL** [reply]
Try another name, "number" is certainly reserve ;)

(Don't know If you can set ID yourself. It's why I show you the validation page, "no id needed")

**Reply by yaj123** [reply]
I tried using different name "selectedNumber"
but still won't display the value;

    public function showEditModal($userID){
        $this->userObject = User::find($userID);
        $this->selectedNumber1 = $this -> userObject -> number;
        $this->selectedUsername = $this -> userObject -> username;
        $this->selectedName = $this -> userObject -> name;
        $this->selectedLocation = $this -> userObject -> location;
        $this->selectedPosition = $this -> userObject -> position;
        $this->mount();
        dd($this->selectedNumber);
       
    }
    
    public function mount()
    {
        $this->selectedNumber = $this->selectedNumber1;
        $this->username = $this->selectedUsername;
        
    }
    
    but in the dd it display that it has a value 
    
![image](https://github.com/livewire/livewire/assets/18673818/b3b611aa-abe4-4a30-89ab-00727a26c0e0)

**Reply by tranKILL** [reply]
I think It's not a good idea to call mount() yourself : https://livewire.laravel.com/docs/lifecycle-hooks#mount

**Reply by yaj123** [reply]
yeah. I was just trying other way around since assigning the $this->selectedNumber in the showEditModal function didn't work. :(
I'm now confused why it won't take the value to the field but it shows in the backend that it has value..

**Reply by Dimimo** [reply]
What does the `userUpdate()` method do? `mount()` is called only once as suggested by @tranKILL,, right before rendering the page. Changing the value of `$selectedNumber` should be done in `userUpdate()` and of course in `showEditModal($userID)` as requested. At the creation of the page, the modal, waiting in the background, can't possibly know what user will be selected. Calling a `mount()` afterwards will be, at best, ignored _(not sure, never tried it)._ 

Updating [properties](https://livewire.laravel.com/docs/properties) can be done with the help of user [actions](https://livewire.laravel.com/docs/actions) and [events](https://livewire.laravel.com/docs/events). Dynamic models can be a bit tricky.

**Reply by yaj123** [reply]
userUpdate method should be the one to save the data from inputs in modal. .so I was trying to copy the data from table data to modal via showEditModal, to put some default value upon pop up of the modal.. 
I'll check this actions and events . .I'm still very new in laravel and livewire..
